include/dmlc/concurrency.h

```diff
@@ -28,7 +28,7 @@ class Spinlock {
     lock_.clear();
   }
 #else
-  Spinlock() : lock_(ATOMIC_FLAG_INIT) {
+  Spinlock() : lock_ ATOMIC_FLAG_INIT {
```

Comment:
?

Reply:
ATOMIC_FLAG_INIT is defined as {false}  so no need for extra parenthesis, as it creates a warning.

Reply:
It's pretty ugly like that

Reply:
Changed it to default initializer

Reply:
Fixed in https://github.com/apache/incubator-mxnet/pull/8870